src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -211,6 +211,9 @@ private static ExternalProjectReference GetExternalProject(PackageSpec rootProje
             // Standard properties
             restoreArgs.ApplyStandardProperties(request);
 
+            // Apply AllowInsecureConnections attribute to RestoreRequest.Project.RestoreMetadata.Sources
```

Comment:
Use Sources from line 170. I think we already handle that deduping there to ensure that the PackageSourceCredentials stays on the PackageSource object.

Reply:
> Use Sources from line 170

I think it is using that one? Do you just mean edit the comment, or the code as well?

Reply:
I don't think it's using it. 
GetEffectiveSources ensures that the PackageSource is the ones that comes form the settings, one that has the PackageSourceCredentials and the likes attached.

Reply:
Thanks for reviewing @nkolev92 and @donnie-msft !
I used the `sources` in line 170, to update the `RestoreRequest.Project.RestoreMetadata.Sources`, but only updated the `AllowInsecureConnections` flag of sources, not updating other attributes.
I just changed it into updating the whole `Sources`.